include/kernels/coarse_op_kernel.cuh

```diff
@@ -786,22 +801,22 @@ namespace quda {
 #pragma unroll
       for (int j = 0; j < TileType::N; j++) {
 
-        if (!isDiagonal) {
+        if (dir == QUDA_IN_PLACE || isDiagonal) {
```

Comment:
Any reason not just to set `isDiagonal = true` always for `QUDA_IN_PLACE`, perhaps simplifying a few of the changes here?

Reply:
Logically no, but `dir == QUDA_IN_PLACE` is compile-time knowledge while `isDiagonal` is runtime, so it's meant to give the compiler more info.

Reply:
Yeh, I figured you'd say that 😉   Although `isDiagonal` isn't `constexpr`, it will be evaluated at compile time for the in-place case since it's unconditionally always true for for in-place.

Reply:
There are some poor logic reasons why I didn't have `dir == QUDA_IN_PLACE` directly imply `isDiagonal` (related to the `-arg.kappa` rescale, see around line 975)---let me know if you'd like me to rework it. It's ultimately a trade-off in where you want your headaches :) 

If you want me to rework this, I'll plug it into my last updates to the MG params PR, if that's alright.

Reply:
No need to do any more updates in this PR with respect to `isDiagonal`, and any such work can be done later.